packages/service-library/src/servicelib/redis.py

```diff
@@ -48,7 +58,25 @@ def __post_init__(self):
             decode_responses=True,
         )
 
-    async def close(self) -> None:
+    async def setup(self) -> None:
+        async for attempt in AsyncRetrying(
+            stop=stop_after_delay(1 * _MINUTE),
+            wait=wait_fixed(_WAIT_SECS),
+            before_sleep=before_sleep_log(logger, logging.WARNING),
+            reraise=True,
+        ):
+            with attempt:
+                if not await self._client.ping():
+                    await self._client.close(close_connection_pool=True)
+                    raise ConnectionError(f"Connection to {self.redis_dsn!r} failed")
+                logger.info(
+                    "Connection to %s succeeded with %s [%s]",
+                    f"redis at {self.redis_dsn=}",
+                    f"{self._client=}",
+                    json.dumps(attempt.retry_state.retry_object.statistics),
+                )
+
+    async def shutdown(self) -> None:
         await self._client.close(close_connection_pool=True)
```

Comment:
Does this raises? All startups and shutdown should doc if they raise exceptions.